service/frontend/workflow_handler.go

```diff
@@ -3718,6 +3828,9 @@ func (wh *WorkflowHandler) PollWorkflowExecutionUpdate(
 	if request.GetUpdateRef() == nil {
 		return nil, errUpdateRefNotSet
 	}
+	if request.UpdateRef.GetWorkflowExecution() == nil {
+		return nil, errExecutionNotSet
+	}
```

Comment:
NOTE: Added this to ensure that the SetAttributes call will not crash, but this is useful/needed anyway.

Reply:
cc @alexshtin